src/PostgREST/lib.sql

```diff
@@ -0,0 +1,18 @@
+-- Make sure that we can create temporary functions if we are in a transaction.
+set transaction read write;
```

Comment:
Oh.. your proposal seems like a really nice idea for maintainability, also more ergonomic development.

But this one seems like a road block(not sure if there's a workaround). A big use case for postgrest is to be able to run in read-only replicas. This transaction level would reject the `create function`:

```sql
BEGIN ISOLATION LEVEL READ COMMITTED READ ONLY;
create or replace function pg_temp.postgrest_numeric_precision(a pg_attribute, t pg_type)
returns integer
language sql
as $$
  select
    information_schema._pg_char_max_length(
      information_schema._pg_truetypid(a.*, t.*),
      information_schema._pg_truetypmod(a.*, t.*)
    )
$$;
ERROR:  25006: cannot execute CREATE FUNCTION in a read-only transaction
LOCATION:  PreventCommandIfReadOnly, utility.c:246
```

Reply:
Oh, I'm sorry. You mentioned:

> default_transaction_read_only could be set to on, but begin read write; or set transaction read write; seems to override that in all cases

Would `set transaction read write` work in read replicas? We would have to test that.